src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+""" show_dampening.py
+    supports commands:
+        * show im dampening
+        * show im dampening {interface}
+"""
+
+# Python
+import re
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Optional
+
+# parser utils
+from genie.libs.parser.utils.common import Common
+
+# =============================================
+# Parser for 'show im dampening'
+# =============================================
+
+class ShowImDampeningSchema(MetaParser):
+    schema = {
+                'interface': {
+                    Any(): {
+                    'protocol': {
+                        Any(): {
+                        'capsulation': str,
+                        'penalty': int,
+                        'suppressed': str,
+                        }
+                    }
+                }
+            }
+        }
+                    
+class ShowImDampening(ShowImDampeningSchema):
+    """ Parser for show im dampening"""
+    
+    cli_command = 'show im dampening'
+
+    def cli(self,output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        result_dict.setdefault('interface', {})
```

Comment:
I see where you are going with this. We need to move this inside the p1.match though. Reason being is we want the parser to return an empty dictionary if nothing parsed as that will raise a SchemaEmptyParserError.

Reply:
Makes sense, I've moved that inside p1.match now. Thanks